java/ql/src/Security/CWE/CWE-200/TempDirLocalInformationDisclosure.ql

```diff
@@ -0,0 +1,213 @@
+/**
+ * @name Temporary Directory Local information disclosure
+ * @description Writing information without explicit permissions to a shared temporary directory may disclose it to other users.
+ * @kind path-problem
+ * @problem.severity warning
+ * @precision very-high
```

Comment:
```suggestion
 * @precision medium
```

Because this query's applicability depends heavily on the nature of data written to the temp directory and/or the nature of the operating environment, this will naturally be FP-heavy